src/software/thunderscope/robot_communication.py

```diff
@@ -44,33 +46,43 @@ def __init__(
         self.motor_control_diagnostics_buffer = ThreadSafeBuffer(1, MotorControl)
         self.power_control_diagnostics_buffer = ThreadSafeBuffer(1, PowerControl)
 
-        self.full_system_proto_unix_io.register_observer(World, self.world_buffer)
-        self.full_system_proto_unix_io.register_observer(
-            PrimitiveSet, self.primitive_buffer
-        )
-        self.full_system_proto_unix_io.register_observer(
-            MotorControl, self.motor_control_diagnostics_buffer
-        )
-        self.full_system_proto_unix_io.register_observer(
-            PowerControl, self.power_control_diagnostics_buffer
-        )
+        self.full_system_running = False
+        self.diagnostics_running = False
+
+        if self.full_system_proto_unix_io:
+            self.full_system_running = True
+            self.full_system_proto_unix_io.register_observer(World, self.world_buffer)
+
+            self.full_system_proto_unix_io.register_observer(
+                PrimitiveSet, self.primitive_buffer
+            )
+
+        if self.diagnostics_proto_unix_io:
+            self.diagnostics_running = True
+            self.diagnostics_proto_unix_io.register_observer(
+                MotorControl, self.motor_control_diagnostics_buffer
+            )
+            self.diagnostics_proto_unix_io.register_observer(
+                PowerControl, self.power_control_diagnostics_buffer
+            )
 
         self.send_estop_state_thread = threading.Thread(target=self.__send_estop_state)
         self.run_thread = threading.Thread(target=self.run)
 
-        try:
-            self.estop_reader = ThreadedEstopReader(
-                self.estop_path, self.estop_buadrate
-            )
-        except Exception:
-            raise Exception("Could not find estop, make sure its plugged in")
+        # try:
+        #     self.estop_reader = ThreadedEstopReader(
+        #         self.estop_path, self.estop_buadrate
+        #     )
+        # except Exception:
+        #     raise Exception("Could not find estop, make sure its plugged in")
 
     def __send_estop_state(self):
-        while True:
-            self.full_system_proto_unix_io.send_proto(
-                EstopState, EstopState(is_playing=self.estop_reader.isEstopPlay())
-            )
-            time.sleep(0.1)
+        print("yea")
+        # while True:
+        #     self.full_system_proto_unix_io.send_proto(
```

Comment:
Should we be checking whether this is `None` or not? (in the case we only run diagnostics and not fullsystem). Potentially same problem with the XProtoListener/Senders lambda functions lower down.